python/pyspark/sql/tests/test_dataframe.py

```diff
@@ -855,6 +855,19 @@ def test_df_show(self):
         with self.assertRaisesRegex(TypeError, "Parameter 'truncate=foo'"):
             df.show(truncate='foo')
 
+    def test_to_pandas_on_spark(self):
```

Comment:
You should run this conditionally. See `python/pyspark/sql/tests/test_arrow.py`.